watertap/flowsheets/full_water_resource_recovery_facility/BSM2.py

```diff
@@ -55,52 +72,136 @@
+    set_operating_conditions(m)
 
-    assert_degrees_of_freedom(m, 0)
-    assert_units_consistent(m)
+    print("---System Initialization---")
     initialize_system(m)
 
-    assert_degrees_of_freedom(m, 0)
-
-    results = solve(m)
-
-    add_costing(m)
+    add_costing(m, has_scalers=True)
+    print("---Costing Initialization---")
     m.fs.costing.initialize()
-    assert_degrees_of_freedom(m, 0)
 
-    results = solve(m)
-    pyo.assert_optimal_termination(results)
+    scale_system(m, has_scalers=has_scalers)
+
+    scaling = pyo.TransformationFactory("core.scale_model")
+    scaled_model = scaling.create_using(m, rename=False)
+
+    dt = DiagnosticsToolbox(scaled_model)
+    print("---Structural Issues---")
+    dt.report_structural_issues()
+
+    # badly_scaled_var_list = iscale.badly_scaled_var_generator(scaled_model, large=1e2, small=1e-2)
+    # print("----------------   badly_scaled_var_list 1   ----------------")
+    # for x in badly_scaled_var_list:
+    #     print(f"{x[0].name}\t{x[0].value}\tsf: {iscale.get_scaling_factor(x[0])}")
+    #
+    # print("--- Scaling Factors ---")
+    # report_scaling_factors(scaled_model, descend_into=True)
+    #
+    print("---Numerical Issues 1---")
+    dt.report_numerical_issues()
+    # dt.display_variables_with_extreme_jacobians()
+    # dt.display_constraints_with_extreme_jacobians()
+    #
+    # print("---SVD 1---")
+    # svd = dt.prepare_svd_toolbox()
+    # svd.display_underdetermined_variables_and_constraints()
+
+    solve(scaled_model)
+
+    # badly_scaled_var_list = iscale.badly_scaled_var_generator(scaled_model, large=1e2, small=1e-2)
+    # print("----------------   badly_scaled_var_list 2   ----------------")
+    # for x in badly_scaled_var_list:
+    #     print(f"{x[0].name}\t{x[0].value}\tsf: {iscale.get_scaling_factor(x[0])}")
+    #
+    # print("--- Scaling Factors 2 ---")
+    # report_scaling_factors(scaled_model, descend_into=True)
+    #
+    # print("---Numerical Issues 2---")
+    # dt.report_numerical_issues()
+    # dt.display_variables_with_extreme_jacobians()
+    # dt.display_constraints_with_extreme_jacobians()
+    # print("---SVD 2---")
+    # svd = dt.prepare_svd_toolbox()
+    # svd.display_underdetermined_variables_and_constraints()
+
+    results = scaling.propagate_solution(scaled_model, m)
 
     print("\n\n=============SIMULATION RESULTS=============\n\n")
-    # display_results(m)
+
+    display_results(m)
     display_costing(m)
 
     setup_optimization(m, reactor_volume_equalities=reactor_volume_equalities)
-    results = solve(m, tee=True)
-    pyo.assert_optimal_termination(results)
-    print("\n\n=============OPTIMIZATION RESULTS=============\n\n")
-    # display_results(m)
-    display_costing(m)
-    display_performance_metrics(m)
+
+    rescale_system(m)
+
+    rescaling = pyo.TransformationFactory("core.scale_model")
+    rescaled_model = rescaling.create_using(m, rename=False)
```

Comment:
@dallan-keylogic Before optimization I was able to reduce the Jacobian Condition number of the solve from 5.502e19 to  4.366e9. However, I'm not exactly sure how scaling should be handled when we after we `setup_optimization`. I imagine we will need to rescale the system since the optimization fails to solve without additional scaling. Am I going about this the right way (creating a `rescale_system` that calls the initial scaling routine and that aims to build on top of it) or is there a better way to do this? For instance, is it possible to update/recommended to just update the initial scaling routine with rather than doing a second scaling transformation?